Update gsoc-2025.md (#4710)
JetBrains/kotlin-web-site: docs/topics/gsoc-2025.md

**Possible mentor**

Jakub Chrzanowski, JetBrains

### Add More Kotlin OpenRewrite Recipes [Medium, 175 hrs]

OpenRewrite is a powerful framework for automating code migrations and refactorings in a structured manner.
While OpenRewrite has strong support for Java, the Kotlin ecosystem would benefit from a more comprehensive
set of OpenRewrite recipes to help developers seamlessly migrate their codebases.

**Expected outcomes**

* Development of new OpenRewrite recipes for Kotlin code migrations

**Skills required (preferred)**

* Kotlin
* OpenRewrite framework
* Java-to-Kotlin migration strategies

**Possible mentor**

Shauvik Roy Choudhary, and the Uber team

### Add BOM Support to Bazel `rules_jvm_external` [Hard, 350 hrs]

> Bazel’s `rules_jvm_external` provides a structured way to declare external Java dependencies,

Reviewer: ```suggestion
Bazel's `rules_jvm_external` provides a structured way to declare external Java dependencies,
```

but it currently lacks proper support for Bill of Materials (BOM) files. 
BOM files are widely used in Maven and Gradle to manage dependencies in a consistent manner
without requiring developers to specify individual versions.
This project aims to enhance `rules_jvm_external` by adding BOM support, allowing developers to use BOM-based dependency resolution within Bazel. 
The project may involve contributing to an existing open-source effort or implementing BOM support directly in `rules_jvm_external`, 
ensuring compatibility with widely used dependency management approaches.

**Expected outcomes**

* Implementation of BOM support in Bazel `rules_jvm_external`
* Improved dependency resolution and usability for Bazel users
* Documentation and examples for using BOM support in Bazel

**Skills required (preferred)**

> * Starlark (Bazel’s scripting language)

Reviewer: ```suggestion
* Starlark (Bazel's scripting language)
```

* Bazel build system
* Dependency resolution strategies

**Possible mentor**

Shauvik Roy Choudhary, and the Uber team

> ### Clean and actionable reporting for Gradle code quality plugins for Kotlin [Easy or Medium, 90 hrs or 175 hrs]

Reviewer: ```suggestion
### Clean and actionable reporting for Gradle code quality plugins for Kotlin [Easy to Medium, 90 hrs to 175 hrs]
```

Gradle recently introduced a new [Problems API](https://docs.gradle.org/current/userguide/reporting_problems.html)
that allows Gradle and third-party plugins to propagate issues and warnings in a unified way. 
This API provides clean and actionable error reporting and more insights into the console output, dedicated HTML reports, 

> and connected observability tools. IDEs such as IntelliJ IDEA or Android Studio also have access to the details via Gradle’s tool integration API, 

Reviewer: ```suggestion
and connected observability tools. IDEs, such as IntelliJ IDEA or Android Studio, also have access to the details via Gradle's tool integration API, 
```

and can show warnings right in the code editorI. Several core features and plugins have already adopted the Problems API: Java compilation, 
dependency resolution errors, deprecation warnings, etc. We want the code quality plugins for Kotlin to adopt this API, too; it would greatly 

> improve the developer experience for 100,000+ Kotlin developers using Gradle.

Reviewer: ```suggestion
improve the developer experience for 100,000+ Kotlin developers using Gradle.

```
